crowbar_framework/app/controllers/machines_controller.rb

```diff
@@ -88,10 +88,19 @@ def role
 
   add_help(:rename, [:id])
   def rename
-    @machine.alias = params[:alias]
+    begin
+      @machine.alias = params[:alias]
+    rescue StandardError => e
```

Comment:
well, that catches all exceptions specific to what happens in the functions that proceed rename in the stack. And those pretty much deal just with alias validation.